src/librbd/librbd.cc

```diff
@@ -1425,6 +1478,50 @@ extern "C" int rbd_mirror_peer_set_cluster(rados_ioctx_t p, const char *uuid,
   return librbd::mirror_peer_set_cluster(io_ctx, uuid, cluster_name);
 }
 
+extern "C" int rbd_list_cgs(rados_ioctx_t p, char *names, size_t *size)
+{
+  librados::IoCtx io_ctx;
+  librados::IoCtx::from_rados_ioctx_t(p, io_ctx);
+  TracepointProvider::initialize<tracepoint_traits>(get_cct(io_ctx));
+  tracepoint(librbd, list_cgs_enter, io_ctx.get_pool_name().c_str(), io_ctx.get_id());
+
+  vector<string> cpp_names;
+  int r = librbd::list(io_ctx, cpp_names);
+
+  if (r == -ENOENT) {
+    tracepoint(librbd, list_cgs_exit, 0, *size);
+    return 0;
```

Comment:
Shouldn't you set `*size = 0` and `*name = '\0'` befor returning here? Or may be just set `r = 0` and allow it to go down?